### PR TITLE
Add cadc-tap 

### PR DESCRIPTION
**Pull request by emeraldTree24**
A sub-package that does gets the schema, does queries, creates tables, deletes tables, creates indices and loads content through the command line

### CODE REVIEW

File: cadctap/cadctap/youcat.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+                        'FITSTable': 'application/fits'}
+
+
+class YoucatClient(object):
+    """Class to do CRUD + visitor actions on a CAOM2 collection repo."""
+
+    def __init__(self, subject, resource_id=DEFAULT_RESOURCE_ID,
+                 host=None, agent=None):
+        """
+        Instance of a CAOM2RepoClient
```
Comment (brianmajor) [inline cadctap/cadctap/youcat.py]: wrong class

File: cadctap/cadctap/youcat.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        :param subject: the subject performing the action
+        :type cadcutils.auth.Subject
+        :param resource_id: the resource ID of the service
+        :param host: Host for the caom2repo service
+        :param agent: The name of the agent (to be used in server logging)
+        """
+        self.logger = logging.getLogger('CAOM2RepoClient')
```
Comment (brianmajor) [inline cadctap/cadctap/youcat.py]: wrong class

File: cadctap/cadctap/youcat.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        if result.status_code == 303:
+            job_url = result.headers['Location']
+            if not job_url:
+                raise RuntimeError(
+                    'table update job location missing in response')
+            # start the job
+            self._tap_client.post('{}/phase'.format(job_url),
+                                  data={'PHASE': 'RUN'})
+            # start polling
+            short_waits = 10
```
Comment (brianmajor) [inline cadctap/cadctap/youcat.py]: Doesn't look like short_waits is used...

File: cadctap/cadctap/core.py [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+        nargs='?',
+        help='ADQL query to run, format is a string with quotes around it, '
+             'for example "SELECT observationURI FROM caom2.Observation"')
+    options_parser.add_argument(
+        '-i', '--input-file',
+        default=None,
+        help='read query string from file (default is from STDIN),'
+             ' location of file')
+    query_parser.add_argument(
+        '-a', '--async-job',
```
[the remark this answers is not in the document]
Comment (andamian) [reply cadctap/cadctap/core.py]: No support for this so we'll comment it out.

File: cadctap/cadctap/core.py [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+
+# make the stream bar show up on stdout
+progress.STREAM = sys.stdout
+
+__all__ = ['CadcTapClient']
+
+# IVOA dateformat
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%f"
+# resource ID for info
+DEFAULT_RESOURCE_ID = 'ivo://cadc.nrc.ca/youcat'
```
[the remark this answers is not in the document]
Comment (andamian) [reply cadctap/cadctap/core.py]: This is for Stephen, so implicitly for youcat. It's a pre-release, so we can change our minds later.

File: cadctap/cadctap/tests/data/help_query.txt [hunk 6]
```diff
@@ -0,0 +1,55 @@
+usage: cadc-tap query [-h]
```
[the remark this answers is not in the document]
Comment (emeraldTree24) [reply cadctap/cadctap/tests/data/help_query.txt]: qtap is going to be in a later release

File: cadctap/cadctap/youcat.py (hunk 3 above)
Comment (emeraldTree24) [reply cadctap/cadctap/youcat.py]: I think wait=1 while short_waits > 0 then when short_waits=0, wait=30 so when it does the tap_client.get it waits for however long wait is.

File: cadctap/README.rst [hunk 7]
```diff
@@ -0,0 +1,9 @@
+cadcdata 
+========
+
+.. image:: https://img.shields.io/pypi/v/cadcdata.svg   
+    :target: https://pypi.python.org/pypi/cadcdata
+
+Canadian Astronomy Data Centre - data access
+
+access library and client for astronomical data hosted at the Canadian Astronomy Data Centre
```
[the remark this answers is not in the document]
Comment (emeraldTree24) [reply cadctap/README.rst]: Changed words but the urls for the images aren't going to exists yet so when they are created they can be changed

File: cadctap/cadctap/__init__.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+Method 1. is the recommended method as it does not required forking external
+processes and also allows trapping the exceptions and reacting according to the
+type of the error. Method 2 also works but the sys.exit needs to be trapped in
+order to prevent the script from quiting. Method 3, while simple, must rely on
+inter processes communication to determine the result of running the command.
+
+"""
+
+from .core import *   # noqa
```
[the remark this answers is not in the document]
Comment (emeraldTree24) [reply cadctap/cadctap/__init__.py]: Update doc

File: cadctap/cadctap/youcat.py [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+
+# allowed file formats for load
+ALLOWED_CONTENT_TYPES = {'tsv': 'text/tab-separated-values', 'csv': 'text/csv'}
+ALLOWED_TB_DEF_TYPES = {'VOSITable': 'text/xml',
+                        'VOTable': 'application/x-votable+xml',
+                        'FITSTable': 'application/fits'}
+
+
+class YoucatClient(object):
+    """Class to do CRUD + visitor actions on a CAOM2 collection repo."""
```
[the remark this answers is not in the document]
Comment (emeraldTree24) [reply cadctap/cadctap/youcat.py]: Updated

File: cadctap/cadctap/youcat.py (hunk 1 above)
Comment (emeraldTree24) [reply cadctap/cadctap/youcat.py]: Updated

File: cadctap/cadctap/youcat.py (hunk 2 above)
Comment (emeraldTree24) [reply cadctap/cadctap/youcat.py]: Updated

**Pull request merged**